apis/kubernetes/v1alpha1/kubernetescluster_types.go

```diff
@@ -289,7 +292,7 @@ type KubernetesClusterMaintenancePolicyObservation struct {
 type KubernetesStatus struct {
 	// A string indicating the current status of the node.
 	// +kubebuilder:validation:Optional
-	State string `json:"state,omitempty"`
+	State KubernetesStateType `json:"state,omitempty"`
```

Comment:
nitpick: I typically will omit something like `Type` as it can be inferred by the identifier being a `type`, but feel free to ignore if you are not aligned
```suggestion
	State ClusterState `json:"state,omitempty"`
```

Reply:
Thanks! I updated it to be `KubernetesState` instead of `ClusterState` since it's also used to describe other resources such as the nodes and not just the cluster. But I do like the idea of not having type at the end.